io/src/main/scala/sbt/io/Hash.scala

```diff
@@ -52,7 +52,18 @@ object Hash {
   def apply(as: Array[Byte]): Array[Byte] = apply(new ByteArrayInputStream(as))
 
   /** Calculates the SHA-1 hash of the given file.*/
-  def apply(file: File): Array[Byte] = Using.fileInputStream(file)(apply)
+  def apply(file: File): Array[Byte] = {
+    val key = (file, file.length, file.lastModified)
+    cache.get(key) match {
+      case Some(hit) =>
+        hit.clone
+      case None =>
+        val result = Using.fileInputStream(file)(apply)
+        cache += (key -> result.clone)
+        result
+    }
+  }
+  var cache: Map[(File, Long, Long), Array[Byte]] = Map.empty
```

Comment:
This should be `private`, right? Also could you document the intended properties of the cache key, for posterity.

Reply:
Ah, I see.